lib/omniauth/strategies/openid_connect.rb

```diff
@@ -313,14 +310,29 @@ def id_token_callback_phase
       end
 
       def valid_response_type?
-        return true if params.key?(options.response_type)
+        return true if params.key?(response_type)
 
-        error_attrs = RESPONSE_TYPE_EXCEPTIONS[options.response_type.to_s]
+        error_attrs = RESPONSE_TYPE_EXCEPTIONS[response_type]
         fail!(error_attrs[:key], error_attrs[:exception_class].new(params['error']))
 
         false
       end
 
+      def response_type
+        @response_type ||= options.response_type.to_s
+      end
+
+      def id_token_response?
+        Array(response_type).include?('id_token')
```

Comment:
Array from a string?

Reply:
This is just rewrite of your own code https://github.com/m0n9oose/omniauth_openid_connect/pull/34/files#diff-8467ba16957a2513a82eb9fe781131dfR186

Reply:
No, it's not. I use `options.response_type` and you use `response_type` method which returns a string.

Reply:
And? Yes, we cast response type to string to make comparisons consistent. How does it correspond to your first question about an array?

Reply:
How it is going to work?
```ruby
[6] pry(main)> Array(['id_token', 'code'].to_s).include?('id_token')
=> false
```

Reply:
> How it is going to work?
> 
> ```ruby
> [6] pry(main)> Array(['id_token', 'code'].to_s).include?('id_token')
> => false
> ```

Ah, okay, I see what you mean. Good catch, thanks